sklearn/utils/_param_validation.py

```diff
@@ -506,6 +534,34 @@ def __str__(self):
         )
 
 
+class _MissingValues(_Constraint):
+    """Helper constraint for the `missing_values` parameters.
+
+    Convenience for
+    [Integral, Real, str, _NoneConstraint, _NanConstraint, _PandasNAConstraint]
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._constraints = [
+            _InstancesOf(Integral),
+            _InstancesOf(Real),
```

Comment:
Technically, Real includes np.nan, so the nan constraint is not necessary here. I think it's fine to keep it to be explicit but maybe we should make the Real constraint it an ``Interval(Real, None, None, closed="both")`` to not overlap with np.nan.
I still think creating a nan constraint is useful because there are some places where we need it and we just set a less tight constraint for now.

Reply:
I will add the following with a comment.